infinispan/src/main/java/org/hibernate/search/infinispan/impl/InfinispanDirectoryProvider.java

```diff
@@ -129,8 +136,30 @@ public void start(DirectoryBasedIndexManager indexManager) {
 		log.debugf( "Initialized Infinispan index: '%s'", directoryProviderName );
 	}
 
+	private void augmentDirectoryWithAsyncDeletes(org.infinispan.lucene.directory.BuildContext directoryBuildContext) {
```

Comment:
Ok sounds like a reasonable plan. I'm totally unable to test this though, did you try this built and integrated with Infinispan 6? (I'm ok to take your word on this, unless you have a great idea to test it).